libtransmission/announce-list.cc

```diff
@@ -225,6 +225,11 @@ bool tr_announce_list::canAdd(tr_url_parsed_t const& announce)
 
 bool tr_announce_list::save(std::string const& torrent_file, tr_error** error) const
 {
+    //we process the magnet file in tr_torrent::setTrackerList
+    if (tr_strvEndsWith(torrent_file, ".magnet"sv)) {
+        return true;
+    }
+    
```

Comment:
This PR makes me realize that `tr_announce_list` has a pre-existing problem with being too tightly coupled to a .torrent file. `tr_announce_list` shouldn't need to know what the internals of a `.torrent` file look like. That's the origin of the problem that's coming to a head here, which is that it _also_ shouldn't have to know about magnet files.

For now, I'd prefer if these five lines were removed to minimize the dependencies and that the caller checks for metainfo before calling save(). I'll try and clean up announce-list, but it's out-of-scope for this PR

```suggestion
```